### PR TITLE
BlockMetadataResponse 2.0 update

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/types/api/core/response.rs [hunk 1]
```diff
@@ -118,17 +119,35 @@ pub struct SubmitBlockResponse {
...
-pub enum LedgerInclusionState {
-    Conflicting,
-    Included,
-    NoTransaction,
+pub enum BlockTransactionState {
+    // Stored but not confirmed or contains not yet included transaction.
+    Pending,
+    // Confirmed with the first level of knowledge.
+    Confirmed,
+    // Included and cannot be reverted anymore.
```
Comment (DaughterOfMars) [inline sdk/src/types/api/core/response.rs:134]:
```suggestion
    // Included and can no longer be reverted.
```

File: sdk/src/types/api/core/response.rs [hunk 2]
```diff
@@ -141,22 +160,21 @@ pub enum LedgerInclusionState {
 )]
 pub struct BlockMetadataResponse {
     pub block_id: BlockId,
-    pub parents: Vec<BlockId>,
-    pub is_solid: bool,
+    pub strong_parents: Vec<BlockId>,
     #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
```
Comment (DaughterOfMars) [inline sdk/src/types/api/core/response.rs:164]: Can we add `default` to all of these?

File: sdk/src/types/block/semantic.rs [hunk 3]
```diff
@@ -45,12 +45,15 @@ impl std::error::Error for ConflictError {}
...
 #[packable(unpack_error = ConflictError)]
 #[packable(tag_type = u8, with_error = ConflictError::InvalidConflict)]
+// TODO may want to rename this to TransactionStateReason
 pub enum ConflictReason {
     /// The block has no conflict.
     None = 0,
     /// The referenced Utxo was already spent.
     InputUtxoAlreadySpent = 1,
     /// The referenced Utxo was already spent while confirming this milestone.
+    /// TODO weird
```
Comment (DaughterOfMars) [inline sdk/src/types/block/semantic.rs:55]: Weird?

Comment (kwek20) [reply sdk/src/types/block/semantic.rs:55]: very weird!

**Pull request closed without merging**